src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -910,8 +910,16 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 			if (!result.matches(editor)) {
 				const targetGroups = editor.hasCapability(EditorInputCapabilities.Untitled) ? this.editorGroupService.groups.map(group => group.id) /* untitled replaces across all groups */ : [groupId];
 				for (const targetGroup of targetGroups) {
+					const resolvedEditor = await this.editorResolverService.resolveEditor({ resource: result.resource, }, targetGroup);
 					const group = this.editorGroupService.getGroup(targetGroup);
-					await group?.replaceEditors([{ editor, replacement: result, options: editorOptions }]);
+					if (resolvedEditor !== ResolvedStatus.ABORT && isEditorInputWithOptionsAndGroup(resolvedEditor)) {
+						// dispose the previous result before replacing with the resolved editor
+						saveResults.pop()?.dispose();
```

Comment:
@lramos15 I am not so sure about this `dispose` call on the editor input. We typically never call `dispose` on editor inputs because the lifecycle may be different depending on the kind of input. For example, all file editor inputs are shared across all groups.

Why is this `dispose` call needed?

Reply:
Well normally we replace `editor` with `result` but here we are replacing `editor` with `resolvedEditor` meaning the editor input returned from save (`result`) isn't being used. So I dispose of it and add the resolved editor as the result instead.

Reply:
Hm yeah, I think this would resolve automatically once `save` / `saveAs` no longer returns a typed editor input.